src/ztex/device_format.c

```diff
@@ -79,15 +76,35 @@ void device_format_init(struct fmt_main *fmt_main,
 	jtr_devices_allow = devices_allow;
 	jtr_verbosity = verbosity;
 
-	struct list_entry *entry;
-	int found_bad_sn = 0;
-	for (entry = devices_allow->head; entry; entry = entry->next)
-		if (!ztex_sn_is_valid(entry->data)) {
-			fprintf(stderr, "Error: bad Serial Number '%s'\n", entry->data);
-			found_bad_sn = 1;
+
+	static int init_conf_devices = 0;
+	if (!init_conf_devices) {
+
+		// Initialize [List.ZTEX:Devices] configuration section.
+		ztex_sn_init_conf_devices();
+
+		// Check -dev command-line option.
+		struct list_entry *entry;
+		int found_error = 0;
+		for (entry = devices_allow->head; entry; entry = entry->next) {
+
+			if (isdigit(entry->data[0]) && (entry->data[1] == 0
+					|| (isdigit(entry->data[1]) && entry->data[2] == 0)) ) {

```

Comment:
ctype macros are defined to operate on `int` and are allowed to malfunction when used on `char`, especially when the 8th bit is set, although typical libc's work around these potential issues. They're also dependent on current locale, which isn't the behavior we want here. I think it's preferable to use comparisons against '0' and '9' here, or maybe use our custom `isdec` function and a length check.

Reply:
OK, agreed.